nemo/collections/nlp/data/language_modeling/megatron/gpt_prompt_learning_dataset.py

```diff
@@ -60,8 +62,7 @@ def __init__(
         add_eos: bool = True,
         for_train: bool = True,
         tokens_to_generate=None,
-        cache_data_path: str = None,  # the cache file
-        load_cache: bool = True,  # whether to load from the cache if it is available
+        max_num_samples=None,
```

Comment:
```suggestion
        max_num_samples: Optional[int] = None,
```
Small nit, could we type the ones that do not have types?